datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -1034,6 +1042,51 @@ private void validateSdxResizeRequest(SdxCluster sdxCluster, String accountId, S
+                        ThreadBasedUserCrnProvider.getUserCrn());
+                if (backupStatus != null && backupStatus.getState() != null
+                        && (backupStatus.getState() == DatalakeOperationStatus.State.STARTED
+                        || backupStatus.getState() == DatalakeOperationStatus.State.IN_PROGRESS
+                        || backupStatus.getState() == DatalakeOperationStatus.State.VALIDATION_FAILED
+                        || backupStatus.getState() == DatalakeOperationStatus.State.VALIDATION_SUCCESSFUL)) {
+                    LOGGER.error("Found a backup in progress on {} with status {}", sdxCluster.getClusterName(), backupStatus.getState());
+                    throw new BadRequestException("SDX cluster is in the process of datalake backup. Resize can not get started.");
+                }
+                LOGGER.info("No backup in progress found on {}. Good to continue.", sdxCluster.getClusterName());
+            } catch (RuntimeException exception) {
+                if (exception.getMessage().contains("Status information for backup operation on datalake")
+                        && exception.getMessage().contains("not found")) {
+                    LOGGER.info("No backup ever happened on {}. Good to continue.", sdxCluster.getClusterName());
+                } else {
+                    throw exception;
+                }
+            }
+            try {
+                DatalakeRestoreStatusResponse restoreStatus = datalakeDrClient.getRestoreStatus(sdxCluster.getClusterName(),
+                        ThreadBasedUserCrnProvider.getUserCrn());
+                if (restoreStatus != null && restoreStatus.getState() != null
+                        && (restoreStatus.getState() == DatalakeOperationStatus.State.STARTED
+                        || restoreStatus.getState() == DatalakeOperationStatus.State.IN_PROGRESS
+                        || restoreStatus.getState() == DatalakeOperationStatus.State.VALIDATION_FAILED
+                        || restoreStatus.getState() == DatalakeOperationStatus.State.VALIDATION_SUCCESSFUL)) {
+                    LOGGER.error("Found a restore in progress on {} with status {}", sdxCluster.getClusterName(), restoreStatus.getState());
+                    throw new BadRequestException("SDX cluster is in the process of datalake restore. Resize can not get started.");
+                }
+                LOGGER.info("No restore in progress found on {}. Good to continue.", sdxCluster.getClusterName());
+            } catch (RuntimeException exception) {
+                if (exception.getMessage().contains("Status information for restore operation on datalake")
+                        && exception.getMessage().contains("not found")) {
+                    LOGGER.info("No restore ever happened on {}. Good to continue.", sdxCluster.getClusterName());
+                } else {
+                    throw exception;
+                }
+            }
+        }
+    }
```

Comment:
Thank you so much @RokolyA94 this looks great! I refactored the code to use `isRunningState `. Same as the above reason, I didn't use the below exception code because that might stop resize where we may not need to. For example, after I added in the exception code many existing test in SdxServiceTest.java fails because it throws a general exception that changes the overall logic of "sdxResize". Unless we know the error message if backupStatus == null (I didnt find an error message for this case in TH though), so I can avoid it stopping resize like what we use later "exception.getMessage().contains". Please let me know what you think!!=) Thank you a lot!
else {LOGGER.info("We do not get resposne from the datalake DR");
throw new RuntimeException();}